src/server/unit_sao.cpp

```diff
@@ -112,6 +122,12 @@ void UnitSAO::sendOutdatedData()
 			m_messages_out.emplace(getId(), true, generateUpdateBonePositionCommand(
 				bone_pos.first, bone_pos.second.X, bone_pos.second.Y));
 		}
+		for (const auto &bone_pos : m_bone_position_unset) {
+			if(!m_bone_position_unset[bone_pos.first]) {
+				m_messages_out.emplace(getId(), true, generateUpdateBonePositionUnsetCommand(bone_pos.first));
+				m_bone_position_unset[bone_pos.first] = true;
+			}
+		}
```

Comment:
I don't see why you have to store a boolean for this.
You could do use a list, add the bone name when it is to be removed (in `unsetBonePosition`) and clear the list here after generating all commands.